### PR TITLE
Extend tests with RAW and CXF_MESSAGE dataFormats

### PR DESCRIPTION
**Pull request by llowinge**

### CODE REVIEW

**Comment by llowinge** [comment]
@ppalaga Can you please check ?

File: integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java [hunk 1]
```diff
@@ -38,16 +47,45 @@ public class CxfSoapClientRoutes extends RouteBuilder {
...
+            "      </ser:add>\n" +
+            "   </soapenv:Body>\n" +
+            "</soapenv:Envelope>";
+
     @Override
     public void configure() {
 
         from("direct:simpleUriBean")
-                .to("cxf:bean:soapClientEndpoint?dataFormat=PAYLOAD");
+                .to("cxf:bean:soapClientEndpoint?dataFormat=POJO");
```
Comment (ppalaga) [inline integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java:65]: Why this change? I thought we are only adding tests here and not changing any thing that worked before. Or this did not work before?

File: integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java [hunk 2]
```diff
@@ -38,16 +47,45 @@ public class CxfSoapClientRoutes extends RouteBuilder {
...
+        from("direct:simpleAddDataFormat")
+                .process(exchange -> {
+                    Map<String, Object> headers = exchange.getIn().getHeaders();
+                    String endpointDataFormat = headers.get("endpointDataFormat").toString();
+                    int[] numbers = exchange.getIn().getBody(int[].class);
+                    String xmlRequest = String.format(MESSAGE_RAW_SIMPLE_ADD, numbers[0], numbers[1]);
+                    if (DataFormat.RAW.name().equals(endpointDataFormat)) {
+                        exchange.getIn().setBody(xmlRequest);
+                    } else if (DataFormat.CXF_MESSAGE.name().equals(endpointDataFormat)) {
+                        try (InputStream is = new ByteArrayInputStream(xmlRequest.getBytes())) {
```
Comment (ppalaga) [inline integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java:80]: The current machine's default encoding can be just anything.
```suggestion
                        try (InputStream is = new ByteArrayInputStream(xmlRequest.getBytes(StandardCharsets.UTF_8))) {
```

**Review by ppalaga — APPROVED** [review]
Some nitpicks inline

File: integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java (hunk 1 above)
Comment (llowinge) [reply integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java:65]: It was working only by accident. With change to @SessionScoped of the `soapClientEndpoint` it has to be `POJO` mode. `PAYLOAD` mode was never used in that case.

File: integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java (hunk 2 above)
Comment (llowinge) [reply integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java:80]: Fixed

File: integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java (hunk 1 above)
Comment (llowinge) [reply integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java:65]: More info in this https://github.com/apache/camel-quarkus/issues/4282#issuecomment-1321228549

**Review by ppalaga — APPROVED** [review]
Thanks, looks good.
